src/nvim/eval.c

```diff
@@ -10199,22 +10199,149 @@ static void f_getcmdwintype(typval_T *argvars, typval_T *rettv)
   rettv->vval.v_string[0] = cmdwin_type;
 }
 
-/*
- * "getcwd()" function
- */
+/// `getcwd([{win}[, {tab}]])` function
+///
+/// Every scope not specified implies the currently selected scope object.
+///
+/// @pre  The arguments must be of type number.
+/// @pre  There may not be more than two arguments.
+/// @pre  An argument may not be -1 if preceding arguments are not all -1.
+///
+/// @post  The return value will be a string.
 static void f_getcwd(typval_T *argvars, typval_T *rettv)
 {
-  char_u      *cwd;
+  // Possible scope of working directory to return.
+  CdScope scope = kCdScopeWindow;
+
+  // Numbers of the scope objects (window, tab) we want the working directory
+  // of. A `-1` means to skip this scope, a `0` means the current object.
+  int scope_number[NUMBER_OF_CD_SCOPES] = {
+    [kCdScopeWindow] = 0,  // Number of window to look at.
+    [kCdScopeTab   ] = 0,  // Number of tab to look at.
+    [kCdScopeGlobal] = 0,  // Number of tab to look at.
+  };
```

Comment:
The only cases where NUMBER_OF_CD_SCOPES is used is for initialization. But in both cases, the array is initialized in the declaration, so NUMBER_OF_CD_SCOPES is redundant.

NUMBER_OF_CD_SCOPES could be eliminated. Since you don't have `default:` cases in the `switch` statements, the compiler will warn us if we add a new `CdScope` enum item.